### PR TITLE
feat(op-reth): Canyon receipts version

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> Note: This is the third PR in the Canyon HF stack. Precedence:
> #5526
> -> #5504
> --> #5503
> ---> `main`

Adds the new `deposit_receipt_version` field as part of the Canyon hardfork. Spec: https://github.com/ethereum-optimism/optimism/blob/develop/specs/deposits.md#deposit-receipt

### CODE REVIEW

File: crates/primitives/src/receipt.rs [hunk 1]
```diff
@@ -203,15 +211,29 @@ impl proptest::arbitrary::Arbitrary for Receipt {
...
-                        _deposit_nonce in any::<Option<u64>>()) -> Receipt
+                        _deposit_nonce in any::<Option<u64>>(),
+                        _deposit_receipt_version in any::<Option<u64>>()) -> Receipt
             {
+                // Only receipts for deposit transactions may contain a deposit nonce
+                #[cfg(feature = "optimism")]
+                let (deposit_nonce, deposit_receipt_version) = if tx_type == TxType::DEPOSIT {
+                    // The deposit receipt version is only present if the deposit nonce is present
+                    let deposit_receipt_version = _deposit_nonce.is_some().then_some(_deposit_receipt_version).flatten();
+                    (_deposit_nonce, deposit_receipt_version)
```
Comment (rkrasiuk) [inline crates/primitives/src/receipt.rs:222]: nit: use `Option::and`

File: crates/primitives/src/receipt.rs [hunk 2]
```diff
@@ -231,8 +253,14 @@ impl<'a> arbitrary::Arbitrary<'a> for Receipt {
 
         // Only receipts for deposit transactions may contain a deposit nonce
         #[cfg(feature = "optimism")]
-        let deposit_nonce =
-            if tx_type == TxType::DEPOSIT { Option::<u64>::arbitrary(u)? } else { None };
+        let (deposit_nonce, deposit_receipt_version) = if tx_type == TxType::DEPOSIT {
+            let deposit_nonce = Option::<u64>::arbitrary(u)?;
+            let deposit_nonce_version =
+                deposit_nonce.is_some().then(|| Option::<u64>::arbitrary(u)).transpose()?.flatten();
```
Comment (rkrasiuk) [inline crates/primitives/src/receipt.rs:259]: nit: `Option::and_then`

**Review by mattsse — APPROVED** [review]
this lgtm, pending @rkrasiuk

**Review by rkrasiuk — APPROVED** [review]

**Comment by mattsse** [comment]
conflict @clabby @rkrasiuk

**Pull request merged**